swagger/src/test/proto/test.proto

```diff
@@ -112,6 +111,11 @@ message PostRequest {
   string excluded = 3 [deprecated = true];
 }
 
+message PostCircularRequest {
+  string string_field = 1 [(test_required) = true];
+  repeated PostCircularRequest circular = 2;
```

Comment:
I agree that this will cause the the condition we are fixing, I would create a separate message instead, like you had in the examples earlier. 

```
message Circular {
  string test = 1;
  repeated Circular circular = 2;
}
```

Then have this message in the request, to be more life-like.

Reply:
@kukido please take another look